### PR TITLE
Update documentation: 'zig init-exe' and 'zig init-lib' are gone

### LINKED ISSUE #18101: zig build advices the use of deprecated commands
### Zig Version

0.12.0-dev.1685+994e19164

### Steps to Reproduce and Observed Output

```
% zig build
info: Initialize a 'build.zig' template file with `zig init-lib` or `zig init-exe`,
or see `zig --help` for more options.
error: No 'build.zig' file found, in the current directory or any parent directories.

% zig init-lib
info: Usage: zig [command] [options]

Commands:

  build            Build project from build.zig
  fetch            Copy a package into global cache and print its hash
  init             Initialize a Zig package in the current directory
...
```

### Expected Output

`init-lib`  and `init-exe` aparently no longer exist, there is only one `init`.

### PR DESCRIPTION
**Pull request by jedisct1**
Fixes #18101

### CODE REVIEW

File: doc/langref.html.in [hunk 1]
```diff
@@ -10851,28 +10851,26 @@ pub fn build(b: *std.Build) void {
...
+    test_step.dependOn(&run_exe_unit_tests.step);
 }
       {#code_end#}
       {#header_close#}
 
       {#header_open|Building a Library#}
-      <p>This <code class="file">build.zig</code> file is automatically generated
-        by <kbd>zig init-lib</kbd>.</p>
+      <p>This <code class="file">build.zig</code> file is a subset of the file
+        automatically generated by the <kbd>zig init</kbd> command.</p>
```
Comment (Vexu) [inline doc/langref.html.in:10873]: `init-exe` and `init-lib` were combined so now this repeats the same build.zig twice.

**Review by Vexu — CHANGES_REQUESTED** [review]

File: doc/langref.html.in (hunk 1 above)
Comment (jedisct1) [reply doc/langref.html.in:10873]: The `build.zig` in the documentation are different, and only contains the bits either to generate a lib, or to generate an executable.

Comment (Vexu) [reply doc/langref.html.in:10873]: Then could you update the documentation to match the output of `zig init`?

Comment (jedisct1) [reply doc/langref.html.in:10873]: But this is the output of `zig init` 😊

My intuition is that in order to create a library, after `zig init`, users will remove the bits relevant to creating an executable, and the other way round. This is exactly what was done here. 

Are you suggesting we merge the "creating a library" and "creating an executable" paragraphs into a single "creating a project" section?

Comment (Vexu) [reply doc/langref.html.in:10873]: > Are you suggesting we merge the "creating a library" and "creating an executable" paragraphs into a single "creating a project" section?

Yes but if you think this is better maybe we should get a third opinion.

**Pull request closed without merging**